iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/plugin/SpringConfigSensor.java

```diff
@@ -63,26 +66,31 @@ public SpringConfigSensor(SonarRuntime sonarRuntime, FileLinesContextFactory fil
     // `sonar-java` plugin, so it's not possible to inject it here.
     // That's why the sensor is hardcoding key and name and not providing a `Language` object.
     super(sonarRuntime, fileLinesContextFactory, noSonarFilter, null);
-    checks = checkFactory.create(SpringConfigExtension.REPOSITORY_KEY);
-    checks.addAnnotatedChecks(SpringConfigCheckList.checks());
+
+    springConfigChecks = checkFactory.create(SpringConfigExtension.JAVA_CONFIG_REPOSITORY_KEY);
+    springConfigChecks.addAnnotatedChecks(SpringConfigCheckList.javaConfigChecks());
```

Comment:
Yes it's needed. Here it's not about registering rules, but basically to get all activeRules from the corresponding repository (java) that are also in the `SpringConfigCheckList.javaConfigChecks()`

Reply:
This one is for `javaconfig`. For `java` repository are lines 73-74

Reply:
Ah ok. So right now it doesn't have any effect because we have no javaConfigChecks. Basically it returns an empty Checks<IacCheck>.
But it doesn't hurt to have it inside so we won't forget it later when adding the first javConfig rule. (And there we need it)

Reply:
Added some comments to clarify the behavior.